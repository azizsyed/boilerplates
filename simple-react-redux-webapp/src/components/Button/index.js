import React from 'react';
import PropTypes from 'prop-types';

const Button = props => <button onClick={props.onClick}>{props.children}</button>;

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
