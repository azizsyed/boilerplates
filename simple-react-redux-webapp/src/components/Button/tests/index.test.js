// Link.react-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { spy } from 'sinon';
import Button from '../';

test('Button', () => {
  const clickSpy = spy();
  const component = renderer.create(
    <Button onClick={clickSpy}>Facebook</Button>
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onClick();

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(clickSpy.called).toBeTruthy();
});
