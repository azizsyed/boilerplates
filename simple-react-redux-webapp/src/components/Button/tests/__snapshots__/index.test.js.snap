// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  onClick={[Function]}
>
  Facebook
</button>
`;

exports[`Button 2`] = `
<button
  onClick={[Function]}
>
  Facebook
</button>
`;
